### DISCUSSION TITLE
columnheader title of nested generic used for accessible name

### DISCUSSION THREAD

**Discussion by eps1lon** [eps1lon/dom-accessibility-api#554] [bug]
Originally reported in https://github.com/testing-library/dom-testing-library/issues/903#issue-815974531 by @nandastone

- `@testing-library/react` version: 9.2.0
- Testing Framework and version: codesandbox.io Jest
- DOM Environment:
  

### Relevant code or config:

```html
<th>
  <span title="Test title">Animal</span>
</th>
```

```js
const columnHeader = screen.getByRole("columnheader", {
  name: /Animal/i
});
```

Test output:

```
Unable to find an accessible element with the role "columnheader" and name `/Animal/i`

Here are the accessible roles:
--------------------------------------------------
  columnheader:

  Name "Test title":
  <th />
```

### What you did:

Attempt to select a column heading by role, using the element text as the accessible name.

### What happened:

`react-testing-library` determines the accessible name as the `title` attribute of the `<span />` element wrapping the element text. Testing across various browsers determines the accessible name should in fact be the element text.

![CleanShot 2021-02-25 at 11 00 38](https://user-images.githubusercontent.com/199204/109087186-befcd480-7758-11eb-9dde-574271d91f0f.png)

![CleanShot 2021-02-25 at 10 59 36](https://user-images.githubusercontent.com/199204/109087188-c2905b80-7758-11eb-992e-1b2372c7a848.png)

### Reproduction:

https://codesandbox.io/s/react-testing-library-demo-forked-1xe1w

### Problem description:

I wrap my column headings in `<span title="" />` to provide tooltips. Instead of selecting the column heading by the text visible to the end user, I need to use tooltip text which is not yet visible.

### Suggested solution:

Mirror browser calculation of the accessible name, or at least provide a way to configure how the `name` is matched in `byRole()`.

**Comment by eps1lon** [comment]
> Mirror browser calculation of the accessible name, or at least provide a way to configure how the name is matched in byRole().

The order of importance is:
1. screen readers
2. [`accname`](https://w3c.github.io/accname/) spec
3. browsers

It might just be that the browser calculation is wrong.

There are multiple ways you can help resolve this issue:
1. Investigate what screen readers announce
2. Investigate how [`accname`](https://w3c.github.io/accname/) would compute this name
3. add a test to https://github.com/eps1lon/dom-accessibility-api/blob/d5ced52f997ce08ff564ddda712c1325b8ab0897/sources/__tests__/accessible-name.js#L63-L134

**Reply by nandastone** [reply]
* The Chrome Screen Reader extension appears to read the visible text i.e. "Animal". 
* VoiceOver reads the title, matching `dom-accessibility-api` behaviour. 

A novice reading of the accname spec appears to support `dom-accessibility-api` behaviour.

Happy to close this.

**Reply by eps1lon** [reply]
@nandastone Appreciate the investigation. Will check with NVDA and JAWS to confirm.

**Reply by eps1lon** [reply]
NVDA+Firefox reads: "Animal"
JAWS+Chrome reads: "Animal"
JAWS+IE11 reads: "Animal"

Will file an issue on `accname` to see what they say. It makes sense to compute `"Animal"` since that's the visible text.
